File: app/src/main/java/com/moggot/findmycarlocation/data/model/route/Route.java
package com.moggot.findmycarlocation.data.model.route;

import com.google.gson.annotations.SerializedName;

import java.util.ArrayList;
import java.util.List;

public class Route {

    @SerializedName("legs")
    private final List<Leg> legs = new ArrayList<>();

    @SerializedName("overview_polyline")
    private OverviewPolyline overviewPolyline;

    public OverviewPolyline getOverviewPolyline() {
        return overviewPolyline;
    }

    public List<Leg> getLegs() {
        return legs;
    }
}

File: app/src/main/java/com/moggot/findmycarlocation/data/model/route/Route.kt
package com.moggot.findmycarlocation.data.model.route

import com.squareup.moshi.Json
import java.util.ArrayList

data class Route(
    @field:Json(name = "legs") val legs: List<Leg> = ArrayList(),
    @field:Json(name = "overview_polyline") val overviewPolyline: OverviewPolyline? = null
)
